controllers/changetierrequest/changetierrequest_controller.go

```diff
@@ -168,6 +168,7 @@ func (r *Reconciler) changeTier(logger logr.Logger, changeTierRequest *toolchain
 				// here we remove the template hash label since it was change for one or all target clusters
 				delete(mur.Labels, nstemplatetier.TemplateTierHashLabelKey(mur.Spec.UserAccounts[i].Spec.NSTemplateSet.TierName))
 				mur.Spec.UserAccounts[i].Spec.NSTemplateSet = newNsTemplateSet
+				mur.Spec.TierName = changeTierRequest.Spec.TierName
```

Comment:
why change it in the for-loop that goes over all UserAccount? wouldn't make more sense to change it outside of the for-loop? The particular `mur.Spec.TierName` field shouldn't be related to UserAccount anymore.